src/CTA.FeatureDetection.Common/Models/Features/FeatureSet.cs

```diff
@@ -67,28 +67,28 @@ public void Add(IEnumerable<Feature> features)
         public void Add(Feature feature)
         {
             var featureType = feature.GetType();
-            if (feature is CompiledFeature)
+            if (feature is CompiledFeature feature1)
             {
                 if (ConfiguredFeatures.Any(f => f.Name == feature.Name))
                 {
                     throw new ArgumentException(string.Format(FeatureNameConflictTemplate, feature.Name, nameof(ConfiguredFeature)));
                 }
 
-                CompiledFeatures.Add((CompiledFeature)feature);
+                CompiledFeatures.Add(feature1);
             }
-            else if (feature is ConfiguredFeature)
+            else if (feature is ConfiguredFeature feature2)
             {
                 if (CompiledFeatures.Any(f => f.Name == feature.Name))
                 {
                     throw new ArgumentException(string.Format(FeatureNameConflictTemplate, feature.Name, nameof(CompiledFeature)));
                 }
 
-                ConfiguredFeatures.Add((ConfiguredFeature)feature);
+                ConfiguredFeatures.Add(feature2);
```

Comment:
nit: can we rename `feature2` to `configuredFeature`?

Reply:
ok